### PR TITLE
lib: add more error context when loading fails

### PR DESCRIPTION
**Pull request by cpu**
Previously the code path that reads from `CertPaths` that configure a dir from the environment lacked helpful context if the dir couldn't be read. Similarly, the code path that read from a `CertPaths` file would add helpful context if `rustls_pemfile` returned an error, but not if the file open failed.

This commit centralizes the job of adding useful context into `CertPaths.load()`. The `rustls_pemfile` specific part in the file path processing can then be removed. This ensures for both dirs and files that if the operation fails there's useful context as to why.

Resolves https://github.com/rustls/rustls-native-certs/issues/116

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -158,12 +158,35 @@ impl CertPaths {
         }
 
         let mut certs = match &self.file {
-            Some(cert_file) => load_pem_certs(cert_file)?,
+            Some(cert_file) => match load_pem_certs(cert_file) {
```
Comment (djc) [inline src/lib.rs:161]: Do you think it's worth abstracting over this to reduce duplication a little bit?

Comment (cpu) [reply src/lib.rs]: :+1: I pushed a revision that lifts out the error construction. It was helpful, turns out I could also simplify the test a little bit too.

**Review by djc — APPROVED** [review]
Great!

**Comment by cpu** [comment]
Feels uninteresting enough that I'm not going to block on a 2nd review. Will iterate for any post-merge feedback :stars:

**Pull request merged**